bin/brew

```diff
@@ -148,6 +151,19 @@ then
   export_homebrew_env_file "/etc/homebrew/brew.env"
 fi
 
+# USER isn't always set so provide a fall back for `brew` and subprocesses.
+export USER="${USER:-$(id -un)}"
+
+if [[ -n "${HOMEBREW_USER-}" && "${HOMEBREW_USER}" != "${USER}" ]]
+then
+  cat >&2 <<ERROR
+Error: \`brew\` is configured to be run by the \`${HOMEBREW_USER}\` user but you are running it as \`${USER}\`!
+Please contact your system administrator for the correct way to run \`brew\`.
+ERROR
+
+  exit 1
+fi
```

Comment:
Might be nicer to delay this check and do it only for relevant subcommands (i.e. you don't need the right user for `--prefix` or `--help`). But kinda want to punt on this for now.

Reply:
Yea, erroring out with `brew --prefix` is a bit much.

I'll close this out for now while reworking it.